lib/collection/src/collection.rs

```diff
@@ -1173,7 +1162,52 @@ impl Collection {
                     }
                     drop(read_shard_holder);
 
-                    // 2 - mark peer as inactive
+                    // 2 - promote local or remote to replica set if needed
+                    if *peer_id == self.this_peer_id {
```

Comment:
This function is already quite big, do you want to maybe move this code block into a separate function?

Reply:
I would prefer to move `handle_replica_changes` out of the collection.rs, which is already close to 2k lines